### PR TITLE
#796. Referenced regulations tag support

### PR DESCRIPTION
**Pull request by madryk**

### CODE REVIEW

File: saos-batch/src/test/java/pl/edu/icm/saos/batch/core/indexer/JudgmentIndexingJobTest.java [hunk 1]
```diff
@@ -97,8 +97,10 @@ public void setUp() throws SolrServerException, IOException, ScriptException, SQ
         EnrichmentTag tag2 = TestInMemoryEnrichmentTagFactory.createReferencedCourtCasesTag(ccJudgments.get(2).getId(), ccJudgments.get(3), ctJudgments.get(0));
         EnrichmentTag tag3 = TestInMemoryEnrichmentTagFactory.createEnrichmentTag(ccJudgments.get(3).getId(), EnrichmentTagTypes.MAX_REFERENCED_MONEY,
                 JsonNormalizer.normalizeJson("{amount:12300.45, text:'123 tys zł 45 gr'}"));
+        EnrichmentTag tag4 = TestInMemoryEnrichmentTagFactory.createEnrichmentTag(ccJudgments.get(3).getId(), EnrichmentTagTypes.REFERENCED_REGULATIONS,
+                JsonNormalizer.normalizeJson("[{journalTitle:'Ustawa', journalNo:162, journalYear:1998, journalEntry:1118, text:'Ustawa (art.103)'}]"));
         
-        enrichmentTagRepository.save(Lists.newArrayList(tag1, tag2, tag3));
+        enrichmentTagRepository.save(Lists.newArrayList(tag1, tag2, tag3)); // TODO save tag4 after resolve of https://github.com/CeON/saos/issues/484
```
Comment (lukdumi) [inline saos-batch/src/test/java/pl/edu/icm/saos/batch/core/indexer/JudgmentIndexingJobTest.java]: Can you explain me what it has to do with 484?

File: saos-batch/src/test/java/pl/edu/icm/saos/batch/core/indexer/JudgmentReindexingJobTest.java [hunk 2]
```diff
@@ -168,7 +168,9 @@ public void judgmentReindexingJob_AFTER_INDEXING() throws Exception {
         EnrichmentTag tag1 = TestInMemoryEnrichmentTagFactory.createReferencedCourtCasesTag(ccJudgments.get(3).getId(), ccJudgments.get(1));
         EnrichmentTag tag2 = TestInMemoryEnrichmentTagFactory.createEnrichmentTag(ccJudgments.get(1).getId(), EnrichmentTagTypes.MAX_REFERENCED_MONEY,
                 JsonNormalizer.normalizeJson("{amount:12300.45, text:'123 tys zł 45 gr'}"));
-        enrichmentTagRepository.save(Lists.newArrayList(tag1, tag2));
+        EnrichmentTag tag3 = TestInMemoryEnrichmentTagFactory.createEnrichmentTag(ccJudgments.get(1).getId(), EnrichmentTagTypes.REFERENCED_REGULATIONS,
+                JsonNormalizer.normalizeJson("[{journalTitle:'Ustawa', journalNo:162, journalYear:1998, journalEntry:1118, text:'Ustawa (art.103)'}]"));
+        enrichmentTagRepository.save(Lists.newArrayList(tag1, tag2)); // TODO save tag3 after resolve of https://github.com/CeON/saos/issues/484
```
Comment (lukdumi) [inline saos-batch/src/test/java/pl/edu/icm/saos/batch/core/indexer/JudgmentReindexingJobTest.java]: same here

File: saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/refregulations/ReferencedRegulationsTagValueItem.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+public class ReferencedRegulationsTagValueItem {
+
+    private String journalTitle;
+    private int journalNo;
+    private int journalYear;
+    private int journalEntry;
+    private String text;
+    
+    
+    public String getJournalTitle() {
```
Comment (lukdumi) [inline saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/refregulations/ReferencedRegulationsTagValueItem.java]: no getters section

File: saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/refregulations/ReferencedRegulationsTagValueItem.java [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    public int getJournalYear() {
+        return journalYear;
+    }
+    public int getJournalEntry() {
+        return journalEntry;
+    }
+    public String getText() {
+        return text;
+    }
+    public void setJournalTitle(String journalTitle) {
```
Comment (lukdumi) [inline saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/refregulations/ReferencedRegulationsTagValueItem.java]: no setters section

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/lawjournal/SimpleLawJournalEntry.java [hunk 5]
```diff
@@ -49,6 +51,10 @@ public String getTitle() {
         return title;
     }
     
+    public String getCode() {
```
Comment (lukdumi) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/lawjournal/SimpleLawJournalEntry.java]: A comment referencing to Judgment#getEntryCode would be helpful.

File: saos-persistence/src/test/java/pl/edu/icm/saos/persistence/service/LawJournalEntryCodeExtractorTest.java [hunk 6]
```diff
@@ -0,0 +1,61 @@
+package pl.edu.icm.saos.persistence.service;
```
Comment (lukdumi) [inline saos-persistence/src/test/java/pl/edu/icm/saos/persistence/service/LawJournalEntryCodeExtractorTest.java]: Shouldn't it be in the search module?

File: saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/refregulations/ReferencedRegulationsTagValueItem.java (hunk 3 above)
Comment (madryk) [inline saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/refregulations/ReferencedRegulationsTagValueItem.java]: :+1:

File: saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/refregulations/ReferencedRegulationsTagValueItem.java (hunk 4 above)
Comment (madryk) [inline saos-enrichment/src/main/java/pl/edu/icm/saos/enrichment/apply/refregulations/ReferencedRegulationsTagValueItem.java]: :+1:

File: saos-webapp/src/main/java/pl/edu/icm/saos/webapp/lawjournal/SimpleLawJournalEntry.java (hunk 5 above)
Comment (madryk) [inline saos-webapp/src/main/java/pl/edu/icm/saos/webapp/lawjournal/SimpleLawJournalEntry.java]: :+1:

**Pull request closed without merging**